dev-tools/mage/benchmark.go

```diff
@@ -0,0 +1,162 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+const (
+	goBenchstat = "golang.org/x/perf/cmd/benchstat@v0.0.0-20230227161431-f7320a6d63e8"
+)
+
+var (
+	benchmarkCount = 8
+)
+
+// Benchmark namespace for mage to group all the related targets under this namespace
+type Benchmark mg.Namespace
+
+// Deps installs required plugins for reading benchmarks results
+func (Benchmark) Deps() error {
+	err := gotool.Install(gotool.Install.Package(goBenchstat))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Run execute the go benchmark tests for this repository, by defining the variable OUTPUT you write the results
+// into a file. Optional you can set BENCH_COUNT to how many benchmark iteration you want to execute, default is 8
+func (Benchmark) Run() error {
+	mg.Deps(Benchmark.Deps)
+	log.Println(">> go Test: Benchmark")
+	outputFile := os.Getenv("OUTPUT")
```

Comment:
just a suggestion.  could we have a default for output that was something like "benchmark-5e71fe7a79.txt", where the 5e71fe7a79 is the output from something like `git log -1 --pretty=format:%h`.  That would make it easy to keep benchmarks from releases and when you are working on code would make it easy to see which benchmark goes with what state of the code.

Reply:
Initially, I did not want to write a file every time but revisiting now seems like a good suggestion. 
I would like to get this merged and I will do follow-up work and will incorporate your suggestion then.